default/python/AI/ProductionAI.py

```diff
@@ -1402,10 +1402,37 @@ def generate_production_orders():
                 if (not prioritized) and (this_priority == PriorityType.PRODUCTION_INVASION):
                     fo.issueRequeueProductionOrder(production_queue.size - 1, 0)  # move to front
         print
+    update_stockpile_use()
     fo.updateProductionQueue()
     _print_production_queue(after_turn=True)
 
 
+def update_stockpile_use():
+    """Decide which elements in the production_queue will be enabled for drawing from the imperial stockpile.  This
+    initial version simply ensures that every resource group with at least one item on the queue has its highest
+    priority item be stockpile-enabled.
+
+    :return: None
+    """
+    # TODO: Do a priority and risk evaluation to decide on enabling stockpile draws
+    empire = fo.getEmpire()
+    production_queue = empire.productionQueue
+    resource_groups = set(tuple(el.key()) for el in empire.planetsWithAvailablePP)
```

Comment:
Does ```empire.planetsWithAvailablePP``` consider supply groups without any PP?

Reply:
My recollection is that I had checked that when writing this code, and the answer was yes.